File: OCH_Tao/ABC/B/271.py
N,Q = map(int,input().split())
L = []
for i in range(N):
  x,*a = map(int,input().split())
  L.append(a)
for i in range(Q):
  S,T = map(int,input().split())
  print(L[S-1][T-1])

File: OCH_Tao/ABC/B/271.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, Q] = input.shift().split(" ").map(x => parseInt(x, 10));
let L = [];
for (let i = 0; i < N; i++) {
  const X = input[i].split(" ");
  L.push(X);
}
for (let i = N; i < N + Q; i++) {
  const [S, T] = input[i].split(" ").map(x => parseInt(x, 10));
  console.log(L[S - 1][T]);
}